lib/measures/tbd/measure.rb

```diff
@@ -15,6 +15,36 @@
   require_relative "resources/model.rb"
   require_relative "resources/transformation.rb"
   require_relative "resources/version.rb"
+  require_relative "resources/log.rb"
+end
+
+# TBD exit strategy when faced with warnings/errors. The generated file targets
+# a design context ( >= WARN ) ... change TBD log_level for debugging purposes.
+# By default, log_status is set below DEBUG while log_level is set @WARN.
+def exitTBD(runner, out_dir, io)
+  # TBD.set_log_level(TBD::DEBUG)
+  unless TBD.logs.empty? || TBD.status < TBD.log_level
+    msgs = []
+    TBD.logs.each do |l|
+      msg = "#{l[:time]} (#{TBD.tag(l[:level])}) #{l[:msg]}"
+      runner.registerError(msg) if l[:level] == TBD::FATAL
+      runner.registerWarning(msg) unless l[:level] == TBD::FATAL
+      msgs << msg
+    end
+    runner.registerWarning("Consult 'tbd.log' - under /files") unless TBD.fatal?
+
+    out_path = File.join(out_dir, "tbd.log")
+    File.open(out_path, "w") do |l|
+      l.write msgs.join("\n")
+      begin
+        l.fsync
+      rescue StandardError
+        l.flush
+      end
+    end
+
+    io[:logs] = msgs if io
```

Comment:
@macumber : The log messages (first written out to "tbd.log") are also repeated in the "tbd.out.json". This seems right for 3rd-party apps reading-in TBD JSON output, or for code compliance checks. More practical than dragging 2x files instead of 1x. This now raises the question why even bother generating a "tbd.log" file in the first place, if everything can be stored in a single "tbd.out.json" file? It may even be worth time-stamping/logging "success" messages as well (e.g., the last time TBD info was "successfully" generated). Thoughts?

Reply:
I like the idea of generating a single `tbd.out.json`, that would be easiest for a third party app like SIMEB to parse.  They could check for errors when they open the results.

Reply:
Agree - working on it. Why not take one step further: why not systematically generate a `tbd.out.json` under _/files_ in all circumstances? At a minimum, it would contains a higher level, timestamped success/failure message e.g., "Success, yet raised warnings", warning/error logs if applicable, plus the usual runner [output](https://github.com/rd2/tbd/blob/1ddd663ecb5938c2ff1d81f9c919955b4dce1151/lib/measures/tbd/measure.rb#L131) if successful. I'll update the PR for you to test.

Reply:
The latest commit illustrates most of what's proposed here. Seems to be working well. In a transition state as the logger branch tests still point to the master JSON schema. It still provides a good idea.

Reply:
FYI, no longer in-between states (i.e., _master_ vs _logger_ branches). Fully tested against revised schema.